src/python/pants/backend/python/tasks/pex_build_util.py

```diff
@@ -145,27 +135,98 @@ def resolve_multi(interpreter, requirements, platforms, find_links):
   :return: Map of platform name -> list of :class:`pkg_resources.Distribution` instances needed
            to satisfy the requirements on that platform.
   """
+  deprecated('1.11.0.dev0',
+    'This function has been moved onto the PexBuildUtil class.')
   python_setup = PythonSetup.global_instance()
   python_repos = PythonRepos.global_instance()
-  platforms = platforms or python_setup.platforms
-  find_links = find_links or []
-  distributions = {}
-  fetchers = python_repos.get_fetchers()
-  fetchers.extend(Fetcher([path]) for path in find_links)
-
-  for platform in platforms:
-    requirements_cache_dir = os.path.join(python_setup.resolver_cache_dir,
-                                          str(interpreter.identity))
-    distributions[platform] = resolve(
-      requirements=[req.requirement for req in requirements],
-      interpreter=interpreter,
-      fetchers=fetchers,
-      platform=expand_and_maybe_adjust_platform(interpreter=interpreter, platform=platform),
-      context=python_repos.get_network_context(),
-      cache=requirements_cache_dir,
-      cache_ttl=python_setup.resolver_cache_ttl,
-      allow_prereleases=python_setup.resolver_allow_prereleases,
-      pkg_blacklist=python_setup.resolver_blacklist,
-      use_manylinux=python_setup.use_manylinux)
-
-  return distributions
+  pex_build_util = PexBuildUtil(python_repos, python_setup)
+  return pex_build_util.resolve_multi(interpreter, requirements, platforms, find_links)
+
+
+class PexBuildUtil(object):
```

Comment:
This may have better SOC as a subsystem that declares its own dependencies on `PythonSetup` and `PythonRepos`, but I'm happy enough with this PR's step forward. As you see fit.

Reply:
Hm. I think I'd want to have a bit more definition around what it's responsibilities are first.

Reply:
The responsibilities don't change, being a subsystem just opts in to ~dependency injection and keeps users from having to know the util's details.